tensorrt_llm/_torch/disaggregation/transceiver.py

```
@@ -869,24 +904,57 @@ def check_gen_transfer_status(self, at_least_request_num: Optional[int]):
-            req.state = LlmRequestState.DISAGG_GENERATION_TRANS_COMPLETE
-            session.close()
-            del self._recv_reqs[rid]
-            del self._recv_sessions[rid]
+            completed_reqs[rid] = req
+            try:
+                # transfer_end already stamped at completion detection above.
+                req.set_kv_cache_size(getattr(req, "py_kv_cache_xfer_bytes", 0))
+                if self._need_aux_transfer(req):
+                    self._apply_aux(session, req)
+                self._assert_disagg_history_declared(req)
+            except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)